packages/flutter_tools/lib/src/dart/dependencies.dart

```diff
@@ -2,89 +2,49 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-import 'dart:convert';
+import 'package:analyzer/analyzer.dart';
```

Comment:
This package (`flutter_tools`) already has a dependency on the analyzer to support `flutter analyze`, so this is not bringing in a new dependency to the package.

Was bringing in a new dependency your main concern? Or is there something else you're concerned about with regard to using the analyzer for this particular purpose?